beetsplug/mbsync.py

```diff
@@ -86,13 +176,18 @@ def singletons(self, lib, query, move, pretend, write):
                                item.mb_trackid,
                                item_formatted)
                 continue
-
+            # Clean up obsolete flexible fields
+            if performer_info:
+                for tag in item:
+                    if tag[:6] == 'mbsync' and tag not in track_info:
+                        del item[tag]
             # Apply.
             with lib.transaction():
                 autotag.apply_item_metadata(item, track_info)
+                ui.show_model_changes(item)
```

Comment:
This seems unrelated to the logic change at hand—any chance this was left over from debugging?

Reply:
It is not left from debugging, it is just an unrelated issue: I noticed that mbsync shows the changes for albums but not for singletons, this corrects it.